AppController/djinn.rb

```diff
@@ -1488,6 +1486,13 @@ def set_read_only(read_only, secret)
     return BAD_SECRET_MSG unless valid_secret?(secret)
     return INVALID_REQUEST unless %w(true false).include?(read_only)
 
+    ZKInterface.get_datastore_servers.each { |machine_ip, port|
+      http = Net::HTTP.new(machine_ip, port)
+      request = Net::HTTP::Post.new('/read-only')
```

Comment:
Is there a chance of exception here or timeouts? Should we consider to retry in case something goes bad? Wondering if we can get in a situation were some datastore are not in read only mode.

Reply:
Yes, there's always a chance that the call will timeout or fail. However, that would cause the SOAP request to fail. The user can retry the call since it's idempotent.